agenta-backend/agenta_backend/routers/testset_router.py

```diff
@@ -363,20 +336,20 @@ async def get_testsets(
         if not has_permission:
             error_msg = f"You do not have permission to perform this action. Please contact your organization admin."
             logger.error(error_msg)
-            return JSONResponse(
-                {"detail": error_msg},
```

Comment:
Please put it back this was the issue of the last server error errors we recently had

Reply:
@aybruhm  The best approach is to use a standard response format structure that encompasses both errors and the requested data. This means the endpoint should return the appropriate response model (e.g., a list of testsets) for successful operations. In case of errors, it should raise an exception with the appropriate status code and message.

Here is an example:
Success Response:
```json
{
  "status": "success",
  "data": {
    "testsets": [
      // list of testsets
    ]
  }
}
```
Error Response:
```json
{
  "status": "error",
  "error": {
    "code": 400,
    "message": "Invalid request parameters"
  }
}
```
You can read more about it here https://blog.postman.com/best-practices-for-api-error-handling/

Now regarding the changes I still recommend keeping them as they are for now and focus on migration code to not bring new bugs.